### PR TITLE
Fix code

### PR DESCRIPTION
**Pull request by BohdanBusko**
**STATUS**
READY

**DESCRIPTION**

screenshot from 2018-12-06 12-00-35

screenshot from 2018-12-06 12-00-24

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [x] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs

### CODE REVIEW

File: app/views/account/workspaces/create.js.erb [hunk 1]
```diff
@@ -1,6 +1,11 @@
 <% if @workspace.persisted? %>
-  $('.navbar-workspaces').html(`<%= j render 'account/workspaces/show_workspaces', workspaces: @workspaces %>`);
-  $('.modal').removeClass('is-active');
+  $('.navbar-workspaces').html('<%= j render 'account/workspaces/show_workspaces', workspaces: @workspaces %>');
+  $('.modal .modal-form').html(`
```
Comment (igorkasyanchuk) [inline app/views/account/workspaces/create.js.erb]: DRY CODE!!!

NEXT TIME IF I SEE DUPLICATES IN CODE I'll BE VERY SAD

**Review by igorkasyanchuk — APPROVED** [review]

File: app/javascript/bundles/Dashboard/components/Dashboard/Dashboard.jsx [hunk 2]
```diff
@@ -43,11 +54,11 @@ class Dashboard extends React.Component {
...
-        <div className="box" key="a" data-grid={{i: 'a', x: 0, y: 0, w: 1, h: 2}}>a</div>
-        <div className="box" key="b" data-grid={{i: 'b', x: 1, y: 0, w: 1, h: 2}}>b</div>
-        <div className="box" key="c" data-grid={{i: 'c', x: 2, y: 0, w: 1, h: 2}}>c</div>
-        <div className="box" key="d" data-grid={{i: 'd', x: 3, y: 0, w: 1, h: 2}}>d</div>
-        <div className="box" key="e" data-grid={{i: 'e', x: 4, y: 0, w: 1, h: 2}}>e</div>
+        <div className="box" key="a">a</div>
+        <div className="box" key="b">b</div>
+        <div className="box" key="c">c</div>
+        <div className="box" key="d">d</div>
+        <div className="box" key="e">e</div>
```
Comment (romaslmd) [inline app/javascript/bundles/Dashboard/components/Dashboard/Dashboard.jsx]: what's `a`, `b`, `c`, `d`, `e`?

File: app/models/project.rb [hunk 3]
```diff
@@ -13,6 +13,7 @@
 # Indexes
 #
 #  index_projects_on_deleted_at    (deleted_at)
+#  index_projects_on_name          (name)
```
Comment (romaslmd) [inline app/models/project.rb:16]: why? it may be a pretty big index..

File: app/models/user.rb [hunk 4]
```diff
@@ -120,6 +120,10 @@ def available_projects
     Project.where(workspace_id: available_workspaces.ids)
   end
 
+  def available_tasks
+    Task.where(project_id: available_projects.ids)
```
Comment (romaslmd) [inline app/models/user.rb]: what's wrong w/ `has_many :available_tasks, through: :available_projects` ?

File: app/views/account/workspaces/create.js.erb [hunk 5]
```diff
@@ -1,6 +1,7 @@
 <% if @workspace.persisted? %>
-  $('.navbar-workspaces').html(`<%= j render 'account/workspaces/show_workspaces', workspaces: @workspaces %>`);
-  $('.modal').removeClass('is-active');
+  $('.navbar-workspaces').html('<%= j render 'account/workspaces/show_workspaces', workspaces: @workspaces %>');
+  $('.modal .modal-form').html('<%= j render 'account/workspaces/modal_flash' %>');
+  setTimeout(function(){ $('.modal').removeClass('is-active') }, 1000);
```
Comment (romaslmd) [inline app/views/account/workspaces/create.js.erb:4]: bootstrap modal?
if yes - use bootstrap JS event to close modal

File: db/migrate/20181206084255_add_indexes_for_search_fields.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+class AddIndexesForSearchFields < ActiveRecord::Migration[5.2]
+  def change
+    add_index :workspaces,  :name
+    add_index :projects,    :name
+    add_index :tasks,       :title
```
Comment (romaslmd) [inline db/migrate/20181206084255_add_indexes_for_search_fields.rb:5]: these are pretty heavy indexes.. are they really necessary?

File: app/views/account/workspaces/create.js.erb (hunk 5 above)
Comment (BohdanBusko) [reply app/views/account/workspaces/create.js.erb:4]: no its bulma

File: app/models/project.rb (hunk 3 above)
Comment (BohdanBusko) [reply app/models/project.rb:16]: because i search by name

File: db/migrate/20181206084255_add_indexes_for_search_fields.rb (hunk 6 above)
Comment (BohdanBusko) [reply db/migrate/20181206084255_add_indexes_for_search_fields.rb:5]: yes i use this fields for search

File: app/models/project.rb (hunk 3 above)
Comment (romaslmd) [reply app/models/project.rb:16]: index is not for search, it's for lookup
it's useful for queries like `find_by(name: '') #=> exact match` not `LIKE or ILIKE` partial match

**Pull request merged**